internal/middlewares/auth.go

```diff
@@ -0,0 +1,75 @@
+package middlewares
+
+import (
+	"errors"
+	"log"
+	"net/http"
+
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+)
+
+const authErrorHeader = "Authentication error: "
+
+type serviceRequest struct {
```

Comment:
Can we rename this to serviceCredentials?

Reply:
:+1: